testing/src/test/java/org/motechproject/nms/testing/it/imi/TargetFileServiceBundleIT.java

```diff
@@ -298,6 +304,101 @@ public void createLargeFile() {
         getLogger().debug("Generated {}", tfn.getFileName());
     }
 
+    @Test
+    public void verifyFT151() throws NoSuchAlgorithmException, IOException {
+
+        // To check that target file should contain correct weekID according to LMP of the subscriber.
+
+        Subscriber subscriber1 = new Subscriber(1111111111L, hindi, aa);
+        subscriber1.setLastMenstrualPeriod(DateTime.now().minusDays(125)); // weekId will be W6_1
+        subscriberDataService.create(subscriber1);
+        Subscription subscription = subscriptionService.createSubscription(1111111111L, hindi, pregnancyPack, SubscriptionOrigin.MCTS_IMPORT);
+        subscription.setNeedsWelcomeMessage(false);
+        subscriptionDataService.update(subscription);
+
+        List<String> contents = new ArrayList<>();
+        String line;
+
+        TargetFileNotification tfn = targetFileService.generateTargetFile();
+
+        File targetDir = new File(settingsService.getSettingsFacade().getProperty("imi.local_obd_dir"));
+        File targetFile = new File(targetDir, tfn.getFileName());
+        MessageDigest md = MessageDigest.getInstance("MD5");
+        int recordCount = 0;
+        try (InputStream is = Files.newInputStream(targetFile.toPath());
+             BufferedReader reader = new BufferedReader(new InputStreamReader(is))) {
+            DigestInputStream dis = new DigestInputStream(is, md);
+            while ((line = reader.readLine()) != null) {
+                recordCount++;
+                contents.add(line.split(",")[7]); //column 8 is for weekId
+            }
+        }
+        String md5Checksum = new String(Hex.encodeHex(md.digest()));
+
+        assertEquals((int)tfn.getRecordCount(), recordCount);
```

Comment:
Seems to me you're expecting only one row. Verifying that the TFN matches is good, but I think you should also be verifying that it is equal to the number of records you're expecting, which is 1. Right?